static-site/src/components/ListResources/Showcase.tsx

```diff
@@ -28,26 +34,31 @@ export const Showcase = ({cards, setSelectedFilterOptions}) => {
   )
 }
 
+type ShowcaseTileProps = {
```

Comment:
Re: [suggestion 2/2](https://github.com/nextstrain/nextstrain.org/pull/874#issuecomment-2129012940)

> [@ivan-aksamentov] Also, interfaces don't have to be aliases to anonymous object types:
> 
> ```diff
> -type ShowcaseTileProps = {
> +interface ShowcaseTileProps {
> ```
> 
> P.S. Watch out for differences as well, notably when extending.

Differences are listed in the [TypeScript Handbook](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces), and answers to this [StackOverflow question](https://stackoverflow.com/q/37233735) provide some clear examples.

I have a preference for `type`. Reasons:

1. It ensures all fields are declared in a single declaration. I don't see any reason to allow adding new fields via multiple `interface` declarations and find that behavior unintuitive when looking at the code in the TypeScript Handbook example.
2. It allows renaming primitives such as `type Name = string`. I just pushed an example of how this can be useful and self-documenting: f2e4113dd3e7c472cbcdcbd65b73b32ed0258972
3. The name `type` aligns with what it's doing and what it's used for (**Type**Script)
4. I don't mind the extra `=`, I actually find it intuitive.

That said, I haven't used `interface` much to make other comparisons so I'd be curious to know if there are other reasons you prefer it over `type`.

Reply:
> Is there any downside to thinking/pretending that it does create a new type?

Giving the appearance that a new type is created but not actually creating one means that (to reiterate Ivan's example) it's very easy to put a `type NucSequence = string` value into a `type Address = string` slot and TS will be happy as a clam. An actual new type wouldn't let you mix them up, and can be a useful guardrail against programming errors.

That said, `type` in [Haskell](https://wiki.haskell.org/Type) and [Python](https://docs.python.org/3/library/typing.html#type-aliases) both create aliases too…

Reply:
> That said, type in [Haskell](https://wiki.haskell.org/Type) and [Python](https://docs.python.org/3/library/typing.html#type-aliases) both create aliases too…

Yeah, this is (overly) common and is a common source of errors. Rust does it as well - the following is just fine!

```rust
type CdsName = String;
type GeneName = String;

fn print_gene(name: GeneName) {
    print!("GENE: {name}")
}

fn main() {
    let cds: CdsName = String::from("this is a CDS");
    print_gene(cds);
}
```

Reply:
Thanks for the examples, I understand the issue better now. But it's specific the point on renaming primitives and I don't see a clear suggestion. Should we avoid usage such as f2e4113dd3e7c472cbcdcbd65b73b32ed0258972 by avoiding `type` entirely and using `interface` for object types?

Reply:
My 2¢ would be to generally avoid aliases except in (hopefully rare) cases where a) they make a complex type used lots of places easier to write repeatedly and b) we really don't want a new specific type. Meeting both (a) and (b) would be fairly rare, I think. Aliases like `type GroupName = string` wouldn't meet that bar and we'd instead use the `string` type directly or actually declare a new `GroupName` type.

Reply:
+1 for not bothering with aliases to simple built-in types

Reply:
Ok, I'll drop f2e4113dd3e7c472cbcdcbd65b73b32ed0258972 to get rid of aliases.

The discussion on `type` vs. `interface` for object types is still up in the air – I'll keep `type` in this PR unless I hear another vote for `interface`. I've also started some work (35e004b0a05d437033c880a4f5be743a5ff23975...98fdcc9d9d49e0371237dcb1c7ebe6adf39e544f) which might influence the decision here.

Reply:
> The discussion on type vs. interface for object types is still up in the air 

I don't feel as strongly as Ivan does about the "2-step procedure of (a) creating an anonymous object type and then (b) assigning it to a name, when there's a native construct describing object types exists - interface", but he's written more TS than me so I'd listen to him! As far as I understand, they still don't guard against the mistakes we've been talking about in this thread, e.g. both allow you to do the following bad thing:

```js
{
    interface Gene {name: string};
    interface Cds {name: string};
    const gene1:Gene = {name: "gene1value"}
    const cds1:Cds = {name: "cds1value"}
    const printGene = (gene: Gene): void => console.log(`GENE: ${gene.name}`)
    printGene(gene1)
    printGene(cds1)
}
{    
    type Gene = {name: string};
    type Cds = {name: string};
    const gene1:Gene = {name: "gene1value"}
    const cds1:Cds = {name: "cds1value"}
    const printGene = (gene: Gene): void => console.log(`GENE: ${gene.name}`);
    printGene(gene1)
    printGene(cds1)
}

```

Tangential question: is the above situation always and necessarily the case in a [structural type system](https://en.wikipedia.org/wiki/Structural_type_system)?

> +1 for not bothering with aliases to simple built-in types

I'd bother with them when it makes things clarifying for development (i.e. increases self-documentation), even though the type system doesn't distinguish between them. As in, we don't have to use them systematically but when it's clarifying to do so we can.

Reply:
> > +1 for not bothering with aliases to simple built-in types
> 
> I'd bother with them when it makes things clarifying for development (i.e. increases self-documentation), even though the type system doesn't distinguish between them. As in, we don't have to use them systematically but when it's clarifying to do so we can.

I find 

```typescript
type Gene = string;

// some potentially large amount of semantic distance away, perhaps in another file
const gene: Gene = "ampR";
```

to be significantly less clear than:

```typescript
const gene = "ampR";
```

because I don't have to wonder what's going on inside `Gene`. I think the "clarifying for development" part can generally be conveyed via good variable names (`gene` being a good example of a horrible one...) 

Aliasing simple types adds an additional layer of abstraction and complexity with (to me) minimal benefit over what it replaces.

Reply:
Switched to `interface` in 213995c9b8a189285ec62094009204e0d981f70d